code/ai/aigoals.cpp

```diff
@@ -266,14 +266,20 @@ void ai_remove_ship_goal( ai_info *aip, int index )
 	// ai goal code look
 	Assert ( index >= 0 );			// must have a valid goal
 
+	if (index == aip->active_goal)
```

Comment:
Any particular reason to move this up here?

Reply:
The `goals[index].ai_mode` check needs to be up here, and consequently I thought there was no need to have more than one `(index == aip->active_goal)` check.

Reply:
👍